contracts/leaser/src/migrate.rs

```diff
@@ -2,101 +2,310 @@ use lease::api::MigrateMsg;
 use platform::batch::Batch;
 use sdk::cosmwasm_std::Addr;
 
-use crate::{error::ContractError, result::ContractResult};
+use crate::{msg::MaxLeases, result::ContractResult};
 
-pub fn migrate_leases<I>(leases: I, lease_code_id: u64) -> ContractResult<Batch>
+pub struct Customer<LeaseIter> {
+    customer: Addr,
+    leases: LeaseIter,
+}
+
+#[derive(Default)]
+#[cfg_attr(test, derive(Debug, Eq, PartialEq))]
+pub struct MigrationResult {
+    pub msgs: Batch,
+    pub next_customer: Option<Addr>,
+}
+
+pub type MayCustomer<LI> = ContractResult<Customer<LI>>;
```

Comment:
I'd suggest naming it with `Maybe` as `May` _may_ imply something else.
```suggestion
pub type MaybeCustomer<LI> = ContractResult<Customer<LI>>;
```